modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php

```diff
@@ -123,6 +123,7 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
 
       // Link.
       $link_render_array = [];
+      $link_url = '<none>';
```

Comment:
```suggestion
      $link_url = '';
```

Reply:
```
$ mkdir -p /src
2$ cd /src
3$ cd /src/az-quickstart-scaffolding
4$ /src/az-quickstart-scaffolding/vendor/bin/phpstan analyse --configuration web/profiles/custom/az_quickstart/phpstan.neon web/profiles/custom/az_quickstart
5   0/219 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%  20/219 [▓▓░░░░░░░░░░░░░░░░░░░░░░░░░░]   9%  40/219 [▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░]  18%  60/219 [▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░]  27%  80/219 [▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░]  36% 120/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░]  54% 140/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░]  63% 160/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░]  73% 180/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░]  82% 200/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░]  91% 219/219 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
6
7 ------ ----------------------------------------------------------------------------------- 
8  Line   modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php  
9 ------ ----------------------------------------------------------------------------------- 
10  240    Variable $link_url might not be defined.                                           
11 ------ ----------------------------------------------------------------------------------- 
12
13
14 [ERROR] Found 1 error                                                          
15
16$ exit

```

Reply:
To resolve the error, you need to ensure that the variable $link_url is defined before it is used in the viewElements method. Currently, `$link_url` might not be defined if certain conditions are not met within the foreach loop. See if you can determine which conditions

You can initialize `$link_url` with a default value at the beginning of the loop to ensure it is always defined.

Reply:
You can think of this as setting a default value for the variable before using it.

Reply:
Have lines 241 and 126 match in terms of using `<non>` or `<nolink>` or `''`

Reply:
@tadean Will investigate further

Reply:
> To resolve the error, you need to ensure that the variable `$link_url` is defined before it is used in the viewElements method.
It was used fine before in viewElements without being initialized. See current line [125](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_card/src/Plugin/Field/FieldFormatter/AZCardDefaultFormatter.php#L125). But perhaps it passes ok with line 137 there due to it being locally set with a local either-or setup?

Since it's now being used outside that block (`if ($item->link_title || $item->link_uri)`), I think initializing as `$link_url = '';` would be appropriate, similar to how `$link_render_array = [];` is initialized as empty above it. Then your line 241 will use $link_url if there's anything, or use the `Url::fromRoute('<none>')` if not, meaning $link_url doesn't need to be none as well.

It could also be rearranged to initialize as `$link_url = Url::fromRoute('<none>')` and then set line 241 as just `$link_url`.